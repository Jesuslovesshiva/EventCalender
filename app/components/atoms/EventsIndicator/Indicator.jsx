import classNames from 'classnames'
import PropTypes from 'prop-types'
import {useMemo} from 'react'

import useMemoizedSelector from '@/hooks/use-memoized-selector'
import {selectors} from '@/modules/entities'
import {EVENT_COLORS} from '@/modules/event/results/types'

const Indicator = props => {
  const {type} = useMemoizedSelector(selectors.makeSelectEventEntity, {
    id: props.eventId
  })

  const bgColor = useMemo(() => {
    let lightness = 'medium'
    if (props.isLight) {
      lightness = 'light'
    }
    return `bg-${EVENT_COLORS[type]}-${lightness}`
  }, [type, props.isLight])

  return (
    <div
      className={classNames(
        'max-w-[10px] min-w-[3px] flex-1 rounded-full',
        bgColor
      )}
    />
  )
}

Indicator.propTypes = {
  eventId: PropTypes.string.isRequired,
  isLight: PropTypes.bool
}

Indicator.defaultProps = {
  isLight: false
}

export default Indicator
