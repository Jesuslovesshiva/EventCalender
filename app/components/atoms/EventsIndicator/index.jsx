import classNames from 'classnames'
import PropTypes from 'prop-types'

import Indicator from './Indicator'

const EventsIndicator = props => (
  <div
    className={classNames(
      'flex justify-center h-[3px] gap-[3px] px-px overflow-hidden',
      props.className
    )}
  >
    {props.eventIds.map(eventId => (
      <Indicator key={eventId} eventId={eventId} isLight={props.isLight} />
    ))}
  </div>
)

EventsIndicator.propTypes = {
  className: PropTypes.string,
  eventIds: PropTypes.arrayOf(PropTypes.string),
  isLight: PropTypes.bool
}

EventsIndicator.defaultProps = {
  className: null,
  eventIds: [],
  isLight: false
}

export default EventsIndicator
