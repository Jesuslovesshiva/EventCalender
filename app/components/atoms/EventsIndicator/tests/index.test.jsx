import EVENT_TYPES, {EVENT_COLORS} from '@/modules/event/results/types'
import configureTestStore from '@/utils/test-utils/configure-test-store'
import render from '@/utils/test-utils/custom-render'
import renderInjectors from '@/utils/test-utils/render-injectors'
import setEntities from '@/utils/test-utils/set-entities'

import EventsIndicator from '../index'

const eventIds = ['123']

const {clearDispatch, store, options} = configureTestStore()
const renderComponent = (props = {}) =>
  render(<EventsIndicator eventIds={eventIds} {...props} />, options)

describe('EventsIndicator', () => {
  renderInjectors(null, options)
  setEntities(store, {
    events: {
      [eventIds[0]]: {id: eventIds[0], type: EVENT_TYPES.TOURNAMENT}
    }
  })

  beforeEach(clearDispatch)

  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should show light background color for isLight prop', () => {
    const {container} = renderComponent({isLight: true})
    expect(container.firstChild.firstChild).toHaveClass(
      `bg-${EVENT_COLORS[EVENT_TYPES.TOURNAMENT]}-light`
    )
  })
})
