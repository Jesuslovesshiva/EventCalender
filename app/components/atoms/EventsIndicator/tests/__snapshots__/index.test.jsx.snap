// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventsIndicator should match the snapshot 1`] = `
<div
  class="flex justify-center h-[3px] gap-[3px] px-px overflow-hidden"
>
  <div
    class="max-w-[10px] min-w-[3px] flex-1 rounded-full bg-green-medium"
  />
</div>
`;
